src/provision.js

```diff
@@ -285,42 +285,51 @@ class Provisioner {
     const d = delayer(500);
     const longD = delayer(2000);
 
-    // We want to have a maximum number of attempts
-    let attemptsLeft = forSpawning.length * 2;
-
     // We want to shuffle up the bids so that we don't prioritize
     // any particular worker type
     forSpawning = shuffle.knuthShuffle(forSpawning);
 
-    // We'll only consider the first 400 requests.  Any that are dropped on the
-    // floor will be computed on the next iteration and have an equal chance of
-    // being submitted
-    forSpawning = forSpawning.slice(0, 400);
+    let byRegion = {};
+    for (let x of forSpawning) {
+      assert(x.bid && x.bid.region);
+      let r = x.bid.region;
+      if (!byRegion[r]) {
+        byRegion[r] = [x];
+      } else {
+        byRegion[r].push(x);
+      }
```

Comment:
You don't need this much code... or a mapping from region to requests...
You just need:
```js
let regions = _.uniq(_.map(forSpawning, ['bid', 'region']));
regions.map(region => _.filter(forSpawning, [['bid', 'region'], region]));

await Promise.all(regions.map(async (forSpawning) => {
  forSpawning = forSpawning.slice(0, 400);
  for (let toSpawn of inRegion) {
  ....
}));
```